src/block_ffm.rs

```diff
@@ -480,6 +481,7 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
         fb: &FeatureBuffer,
         pb: &mut PortBuffer,
         caches: &[BlockCache],
+        mask_interactions: bool,
```

Comment:
Instead of passing this here, add it to the creation of the struct with the dropout rate parameter. Will make it way cleaner & it doesn't require such a big blast radius of changes.